lib/iris/tests/unit/analysis/maths/__init__.py

```diff
@@ -159,3 +159,31 @@ def test_partial_mask_not_in_place(self):
 
         self.assertMaskedArrayEqual(com, res.data)
         self.assertIsNot(res, orig_cube)
+
+
+class CubeArithmeticMaskedConstantTestMixin(
+        six.with_metaclass(ABCMeta, object)):
+
+    def test_masked_constant_in_place(self):
+        # Cube in_place arithmetic operation.
+        dtype = np.int64
+        fill_value = 12345
+        dat = np.ma.masked_array(0, 1, dtype)
+        cube = Cube(dat, fill_value=fill_value)
+        res = self.cube_func(cube, 5, in_place=True)
+        self.assertMaskedArrayEqual(np.ma.masked_array(0, 1), res.data)
+        self.assertEqual(dtype, res.dtype)
+        # self.assertEqual(fill_value, res.fill_value)
```

Comment:
Currently the fill value is being lost by maths operations, I don't know whether we need to rectify that?